lib/App/index.js

```diff
@@ -583,33 +595,40 @@ class App {
   _validateFlowCard(card, errorPath, appJson, { levelPublish, levelVerified }) {
     if (Array.isArray(card.args) === false) return;
 
-    let foundFirstDevice = false;
+    const firstDeviceArgument = card.args.find(argument => {
+      const filter = new URLSearchParams(argument.filter);
+      return argument.type === 'device' && (filter.has('driver_id') || filter.has('driverId'));
+    });
 
     // Filter to valid arguments for flow card titles
-    const filteredCardArgs = card.args.filter(arg => {
-      if (foundFirstDevice) return true;
-      if (arg.type !== 'device') return true;
-      if (!arg.filter) return true;
-      const query = querystring.parse(arg.filter);
-      if (!query.driver_id && !query.driverId) return true;
-      foundFirstDevice = true;
-      return false;
+    const filteredCardArgs = card.args.filter(argument => {
+      return argument !== firstDeviceArgument;
     });
 
     if (filteredCardArgs.length === 0) return;
 
+    if (levelVerified) {
+      for (const argument of filteredCardArgs) {
+        if (argument.title === undefined) {
```

Comment:
Why are these checks for `undefined` and not also `null`? Bit odd that someone would set a property to `null` but still possible.

Reply:
AJV will error if any of these are set to `null` , but `undefined` is allowed when a property is not required.